crates/transaction-generator-lib/src/publishing/entry_point_trait.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::publish_util::Package;
+use aptos_sdk::{
+    move_types::{
+        account_address::AccountAddress, identifier::Identifier, language_storage::ModuleId,
+    },
+    types::transaction::{EntryFunction, TransactionPayload},
+};
+use rand::rngs::StdRng;
+
+pub trait PreBuiltPackages: std::fmt::Debug + Sync + Send {
+    fn package_metadata(&self, package_name: &str) -> &[u8];
+    fn package_modules(&self, package_name: &str) -> &[Vec<u8>];
+    fn package_script(&self, package_name: &str) -> Option<&Vec<u8>>;
+}
+
+pub enum MultiSigConfig {
+    None,
+    Random(usize),
+    Publisher,
+    FeePayerPublisher,
+}
+
+/// Automatic arguments function expects (i.e. signer, or multiple signers, etc)
+/// That execution can add before the call.
+#[derive(Debug, Copy, Clone)]
+pub enum AutomaticArgs {
+    None,
+    Signer,
+    SignerAndMultiSig,
+}
+
+pub trait EntryPointTrait: std::fmt::Debug + Sync + Send + CloneEntryPointTrait {
```

Comment:
Why not just `EntryPointTrait: std::fmt::Debug + Sync + Send + Clone`? Why do we need a separate `CloneEntryPointTrait` here?

Reply:
CloneEntryPointTrait is to clone Box of abstract type - something like https://users.rust-lang.org/t/solved-is-it-possible-to-clone-a-boxed-trait-object/1714/7